src/Symfony/Component/HttpKernel/DependencyInjection/RegisterControllerArgumentLocatorsPass.php

```diff
@@ -70,13 +68,12 @@ public function process(ContainerBuilder $container)
             if (!$r = $container->getReflectionClass($class)) {
                 throw new InvalidArgumentException(sprintf('Class "%s" used for service "%s" cannot be found.', $class, $id));
             }
-            $isContainerAware = $r->implementsInterface(ContainerAwareInterface::class) || is_subclass_of($class, AbstractController::class);
 
             // get regular public methods
             $methods = [];
             $arguments = [];
             foreach ($r->getMethods(\ReflectionMethod::IS_PUBLIC) as $r) {
-                if ('setContainer' === $r->name && $isContainerAware) {
+                if ('setContainer' === $r->name) {
```

Comment:
why removing this ? If a class has a `setContainer` method for another reason, skipping it might cause issues.

Reply:
because it feels really bad to have the FrameworkBundle namespace in the HttpKernel component :)
there is zero chance this can be a problem: this pass is about controllers - nobody will even name an action "setContainer"